src/Microsoft.DotNet.VersionTools/BuildManifest/Model/OrchestratedBuildModel.cs

```diff
@@ -0,0 +1,61 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Xml.Linq;
+
+namespace Microsoft.DotNet.VersionTools.BuildManifest.Model
+{
+    public class OrchestratedBuildModel
+    {
+        public OrchestratedBuildModel(BuildIdentity identity)
+        {
+            if (identity == null)
+            {
+                throw new ArgumentNullException(nameof(identity));
+            }
+            if (string.IsNullOrEmpty(identity.BuildId))
+            {
+                throw new ArgumentException(
```

Comment:
With ArgumentNullException/empty on the BuildIdentity constructor this check would be duplicate.

Reply:
Yeah, originally I didn't think Builds should require it, just OrchestratedBuilds, but I'll just pass a placeholder in `PushToBlobFeed` if nothing's provided. This won't be a mainline scenario in any case.